src/org/team1540/robot2017/subsystems/GearMechanism.java

```diff
@@ -1,19 +1,98 @@
-package org.team1540.robot2017.subsystems;
-
-import org.team1540.robot2017.RobotMap;
-
-import com.ctre.CANTalon;
-
-import edu.wpi.first.wpilibj.command.Subsystem;
-
-public class GearMechanism extends Subsystem {
-
-	private final CANTalon gearSliderTalon = new CANTalon(RobotMap.gearSliderTalon);
-//	private final Servo gearRightDeployServo = new Servo(RobotMap.gearServoRightDeploy);
-//	private final Servo gearLeftDeployServo = new Servo(RobotMap.gearServoLeftDeploy);
-	
-	@Override
-	protected void initDefaultCommand() {
-		
-	}
-}
+//package org.team1540.robot2017.subsystems;
+//
+//import org.team1540.robot2017.RobotMap;
+//import org.team1540.robot2017.RobotUtil;
+//
+//import com.ctre.CANTalon;
+//import com.ctre.CANTalon.TalonControlMode;
+//
+//import edu.wpi.first.wpilibj.Counter;
+//import edu.wpi.first.wpilibj.DigitalInput;
+//import edu.wpi.first.wpilibj.Encoder;
+//import edu.wpi.first.wpilibj.Servo;
+//import edu.wpi.first.wpilibj.command.Subsystem;
+//
+//public class GearMechanism extends Subsystem {
+//
+//	private final CANTalon gearSliderTalon = new CANTalon(RobotMap.gearSliderTalon);
+//	private final Servo gearRightDeployServo = new Servo(RobotMap.gearServoRight);
+//	private final Servo gearLeftDeployServo = new Servo(RobotMap.gearServoLeft);
+//	
+//	DigitalInput limitSwitch = new DigitalInput(1);
+//	Counter counter = new Counter(limitSwitch);
+//	private double autoSpeed;
+//	private boolean useJoystick;
+//	private boolean gearLock;
+//	private Encoder sliderEncoder;
+//	
+//	public GearMechanism() {
+//		gearLock = true;
+//		useJoystick = true;
+//		gearSliderTalon.changeControlMode(TalonControlMode.PercentVbus);
+//		autoSpeed = 0;
+//		sliderEncoder = new Encoder(0, 1, false, Encoder.EncodingType.k2X);
```

Comment:
Are you sure about EncodingType.k2X? I have no clue what it should be; this might need some further research if you aren't sure.